packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts

```diff
@@ -35,6 +35,22 @@ describe('profile tests', () => {
     expect(getProfileCredentials_mock).toHaveBeenCalledTimes(0);
   });
 
+  it('should fail to return profiled aws credentials', async () => {
+    const profile_file_contents = '[fake]\nmalformed_key_id=fakeAccessKey\nmalformed_secret_access_key=fakeSecretKey\n'
+    fs_mock.readFileSync.mockImplementationOnce(() => {
+      return profile_file_contents;
+    }).mockImplementationOnce(() => {
+      return profile_file_contents;
+    });
+    const getProfileCredentials_mock = jest.fn(getProfileCredentials);
+    try {
+      const profile_config = getProfiledAwsConfig(context_stub, 'fake');
+    } catch (e) {
+      expect(e.message).toEqual("Profile configuration for 'fake' is invalid: missing aws_access_key_id, aws_secret_access_key");
+    }
```

Comment:
You can use [`expect(...).toThrow(...)`](https://jestjs.io/docs/expect#tothrowerror) here instead of `try...catch`.

Reply:
I tried to use that first but it wouldn't work with the async method for some reason, this was the only way I managed to isolate it

Reply:
Ah, can you try with [`.rejects`](https://jestjs.io/docs/asynchronous).

Reply:
Got it, thanks!